lightning/src/routing/router.rs

```diff
@@ -1221,4 +1598,671 @@ mod tests {
+			fee_base_msat: 0,
+			fee_proportional_millionths: 0,
+			excess_data: Vec::new()
+		});
+
+		// First, let's see if routing works if we have absolutely no idea about the available amount.
+		// In this case, it should be set to 10_000 sats.
+		update_channel(&net_graph_msg_handler, &secp_ctx, &privkeys[0], UnsignedChannelUpdate {
+			chain_hash: genesis_block(Network::Testnet).header.block_hash(),
+			short_channel_id: 3,
+			timestamp: 2,
+			flags: 0,
+			cltv_expiry_delta: 0,
+			htlc_minimum_msat: 0,
+			htlc_maximum_msat: OptionalField::Absent,
+			fee_base_msat: 0,
+			fee_proportional_millionths: 0,
+			excess_data: Vec::new()
+		});
+
+		{
+			// Attempt to route more than is available results in a failure.
+			if let Err(LightningError{err, action: ErrorAction::IgnoreError}) = get_route(&our_id, &net_graph_msg_handler.network_graph.read().unwrap(), &nodes[2], None, &Vec::new(), 10_000_001, 42, Arc::clone(&logger)) {
+				assert_eq!(err, "Failed to find a sufficient route to the given destination");
+			} else { panic!(); }
+		}
+
+		{
+			// Now, attempt to route an exact amount we have should be fine.
+			let route = get_route(&our_id, &net_graph_msg_handler.network_graph.read().unwrap(), &nodes[2], None, &Vec::new(), 10_000_000, 42, Arc::clone(&logger)).unwrap();
+			assert_eq!(route.paths.len(), 1);
+			let path = route.paths.last().unwrap();
+			assert_eq!(path.len(), 2);
+			assert_eq!(path.last().unwrap().pubkey, nodes[2]);
+			assert_eq!(path.last().unwrap().fee_msat, 10_000_000);
+		}
+
+
+		// Now let's see if routing works if we know only htlc_maximum_msat.
+		update_channel(&net_graph_msg_handler, &secp_ctx, &privkeys[0], UnsignedChannelUpdate {
+			chain_hash: genesis_block(Network::Testnet).header.block_hash(),
+			short_channel_id: 3,
+			timestamp: 3,
+			flags: 0,
+			cltv_expiry_delta: 0,
+			htlc_minimum_msat: 0,
+			htlc_maximum_msat: OptionalField::Present(15_000),
+			fee_base_msat: 0,
+			fee_proportional_millionths: 0,
+			excess_data: Vec::new()
+		});
+
+		{
+			// Attempt to route more than is available results in a failure.
+			if let Err(LightningError{err, action: ErrorAction::IgnoreError}) = get_route(&our_id, &net_graph_msg_handler.network_graph.read().unwrap(), &nodes[2], None, &Vec::new(), 15_001, 42, Arc::clone(&logger)) {
+				assert_eq!(err, "Failed to find a sufficient route to the given destination");
+			} else { panic!(); }
+		}
+
+		{
+			// Now, attempt to route an exact amount we have should be fine.
+			let route = get_route(&our_id, &net_graph_msg_handler.network_graph.read().unwrap(), &nodes[2], None, &Vec::new(), 15_000, 42, Arc::clone(&logger)).unwrap();
+			assert_eq!(route.paths.len(), 1);
+			let path = route.paths.last().unwrap();
+			assert_eq!(path.len(), 2);
+			assert_eq!(path.last().unwrap().pubkey, nodes[2]);
+			assert_eq!(path.last().unwrap().fee_msat, 15_000);
+		}
+
+		// // Now let's see if routing works if we know only capacity from the UTXO.
```

Comment:
After a recent change, it's not possible anymore to make net_monitor UTXO-aware in the middle (without reinitializing), so there is no easy way to make it work... Perhaps I'll find a way, or someone suggest it here, but keeping it commented out to allow review for now.